test/performance/observed_concurency_test.go

```diff
@@ -40,17 +39,14 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
-const (
-	concurrency = 5
-)
+const concurrency = 5
 
 // generateTraffic loads the given endpoint with the given concurrency for the given duration.
 // All responses are forwarded to a channel, if given.
-func generateTraffic(client *spoof.SpoofingClient, url string, concurrency int, duration time.Duration, resChannel chan *spoof.Response) (int32, error) {
-	var (
-		group    errgroup.Group
-		requests int32
-	)
+func generateTraffic(client *spoof.SpoofingClient, url string, concurrency int, duration time.Duration, resChannel chan *spoof.Response, logger *logging.BaseLogger) error {
+	var group errgroup.Group
+	// Notify the consumer about the end of the data stream.
+	defer close(resChannel)
```

Comment:
Should we create and return the resChannel here. We close it so this function should own the whole lifecycle.

Reply:
I don't think we can, since this is executed in a go-routine, and the channel is consumed in a reader. :/

Reply:
Gotcha, makes sense. Sorry for not checking better.